java/arcs/core/allocator/Allocator.kt

```diff
@@ -90,39 +95,27 @@ class Allocator(val hostRegistry: HostRegistry) {
      */
     private fun createStorageKey(
         arcId: ArcId,
-        spec: HandleSpec,
         idGenerator: Id.Generator
-    ): StorageKey = when {
-        !isVolatileHandle(spec.tags) -> RamDiskStorageKey(
-            idGenerator.newChildId(arcId, "").toString()
-        )
-        else -> VolatileStorageKey(
+    ): StorageKey = VolatileStorageKey(
             arcId,
             idGenerator.newChildId(arcId, "").toString()
         )
-    }
 
     private fun isVolatileHandle(tags: Set<String>) = tags.contains("volatile")
 
     /**
      * Slice plan into pieces grouped by [ArcHost], each group consisting of a [PlanPartition]
-     * that lists [HandleSpec], [ParticleSpec], and [HandleConnectionSpec] needed for that host.
+     * that lists [ParticleSpec] needed for that host.
      */
     private suspend fun computePartitions(arcId: ArcId, plan: Plan): List<PlanPartition> =
-        plan.handleConnectionSpecs
-            .map { spec -> findArcHostBySpec(spec.particleSpec) to spec }
-            .groupBy({ it.first }, { it.second.particleSpec })
-            // map ArcHost -> List<ParticleSpec> into List<PlanPartition>
-            .map {
-                // find all HandleConnectionSpecs for the given List<ParticleSpec>
-                val handleConnectionSpecs =
-                    plan.handleConnectionSpecs.filter { spec ->
-                        it.value.contains(spec.particleSpec)
-                    }
+        plan.particles
+            .map { spec -> findArcHostBySpec(spec) to spec }
+            .groupBy({ it.first }, { it.second })
```

Comment:
Could probably be reduced to just `groupBy` without the map, by inlining the `findArcHostBySpec(spec)` into `groupBy`'s first arg?